focuspoints.lrdevplugin/PentaxDelegates.lua

```diff
@@ -0,0 +1,180 @@
+--[[
+  Copyright 2016 Whizzbang Inc
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--]]
+
+--[[
+  A collection of delegate functions to be passed into the DefaultPointRenderer
+  for Pentax cameras.
+
+  Notes:
+  * Back focus button sets AFPointsInFocus to 'None' regardless of point
+    selection mode (sucks). AFPointSelected is set correctly with either button.
+
+  2017.03.29 - roguephysicist: works for Pentax K-50 with both phase and contrast detection
+--]]
+
+local LrStringUtils = import "LrStringUtils"
+local LrErrors = import 'LrErrors'
+require "Utils"
+
+-- LrErrors.throwUserError(result)
+
+
+PentaxDelegates = {}
+
+PentaxDelegates.focusPointsMap = nil
+PentaxDelegates.focusPointDimen = nil
+PentaxDelegates.metaKeyFocusMode = {"Focus Mode"}
+
+function PentaxDelegates.getAfPoints(photo, metaData)
+  local focusMode = ExifUtils.findFirstMatchingValue(metaData, PentaxDelegates.metaKeyFocusMode)
+  if (string.find(focusMode, 'AF%-S') ~= nil) then
+    results = PentaxDelegates.getAfPointsPhase(photo, metaData)
+  elseif (string.find(focusMode, 'Contrast%-detect') ~= nil) then
+    results = PentaxDelegates.getAfPointsContrast(photo, metaData)
+  end
+  return results
+end
+
+--[[
+-- photo - the photo LR object
+-- metaData - the metadata as read by exiftool
+--]]
+function PentaxDelegates.getAfPointsPhase(photo, metaData)
+  local afPointsSelected = ExifUtils.findFirstMatchingValue(metaData, { "AF Point Selected" })
+  if afPointsSelected == nil then
+    afPointsSelected = {}
+  else
+    afPointsSelected = splitTrim(afPointsSelected, ";") -- pentax separates with ';'
+  end
+  local afPointsInFocus = ExifUtils.findFirstMatchingValue(metaData, { "AF Points In Focus" })
+  if afPointsInFocus == nil then
+    afPointsInFocus = {}
+  else
+    afPointsInFocus = splitTrim(afPointsInFocus, ",")
+    afPointsInFocus = PentaxDelegates.fixCenter(afPointsInFocus)
+  end
+    
+  local result = {
+    pointTemplates = DefaultDelegates.pointTemplates,
+    points = {}
+  }
+
+  for key,value in pairs(PentaxDelegates.focusPointsMap) do
+    local x = PentaxDelegates.focusPointsMap[key][1]
```

Comment:
Thinking out loud...
this is a bit of difference in feature. It looks to me like the code is adding all focus points even when inactive. It's a cool feature, but I dont think the other delegates do this. I know the DefaultDelegate doesn't do that. Let's grab some canon images and see if it does it. If it's a change in how the app, handles, I'm not sure what do to with it right now. Perhaps a setting button to enable/disable it? Dont remove this, leave as-is for now. Feel free to give your thoughts.